homeassistant/components/philips_js/media_player.py

```diff
@@ -192,7 +192,7 @@ def is_volume_muted(self):
 
     def turn_on(self):
         """Turn on the device."""
-        if self._on_script:
+        if self._on_script and not self._tv.on:
```

Comment:
We shouldn't put this as part of the integration but it's up to the caller to check.

Reply:
Because if the state is out of sync, the user can no longer turn it on .

Reply:
State will get in sync. The lib will set on to false as soon as soon it can't connect anymore.

Reply:
the script allows you to use condition if you need guard this feature

Reply:
Ok. I'll close. Will se if i update docs with an example for that.